OpenRA.Game/Network/OrderManager.cs

```diff
@@ -131,8 +131,12 @@ public void TickImmediate()
 						return;
 
 					var frame = BitConverter.ToInt32(packet, 0);
-					if (packet.Length == 5 && packet[4] == (byte)OrderType.Disconnect)
-						pendingPackets.Remove(clientId);
+					if (packet.Length == Order.DisconnectOrderLength + 4 && packet[4] == (byte)OrderType.Disconnect)
```

Comment:
It's weird that the server code treats `OrderType` as the first byte in the data payload rather than as part of the framing. Fixing this isn't trivial because the `OrderType` handling is split half between the network code and half inside `Order`, so best that we leave this to a future PR.